src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -78,21 +96,25 @@ public boolean isEmpty()
         return constraints.isEmpty();
     }
 
-
     @Override
-    public void validate(ColumnMetadata columnMetadata, TableMetadata tableMetadata) throws InvalidConstraintDefinitionException
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
     {
+        if (complexTypes.contains(columnMetadata.type))
+        {
+            throw new InvalidConstraintDefinitionException("Complex types do not support constraints");
+        }
```

Comment:
I think you want to check with the class, not the `AbstractType` object. I also enriched the error message a bit

```suggestion
        if (complexTypes.contains(columnMetadata.type.getClass()))
        {
            throw new InvalidConstraintDefinitionException("Constraint cannot be defined on column '" 
                                                           + columnMetadata.name + "' with type: " 
                                                           + columnMetadata.type.asCQL3Type());
        }
```